src/components/UploadScreen.tsx

```diff
@@ -198,9 +233,52 @@ const UploadScreen = () => {
       formatDescription: validation.formatDescription,
       availableStates: validation.availableStates,
       selectedState: validation.availableStates?.includes('diff') ? 'diff' : validation.availableStates?.[0],
-    });
+    };
+
+    setUploadState(newUploadState);
+
+    if (validation.isValid) {
+      const dataToStore: StoredFileData = {
+        content: validation.content,
+        format: validation.format,
+        formatDescription: validation.formatDescription,
+        availableStates: validation.availableStates,
+        selectedState: validation.availableStates?.includes('diff') ? 'diff' : validation.availableStates?.[0],
+      };
+      sessionStorage.setItem(SESSION_STORAGE_KEY, JSON.stringify(dataToStore));
+    }
```

Comment:
This code seems duplicated couple of times, couldn't it be done via `setUploadState`? I.e. you could have a callback `handleUploadState` that would deal with adding to session storage and calling the original `setUploadState`.